backend/src/main/java/com/example/dontwastetomuch/game/GameController.java

```diff
@@ -158,4 +157,23 @@ public ResponseEntity<CommunityStatsDTO> getOneCommunityGame(@PathVariable Strin
             return new ResponseEntity(e.getMessage(), HttpStatus.NOT_FOUND);
         }
     }
+
+    @GetMapping("/edit/{gameId}")
+    public ResponseEntity<Game> getOneGameToEdit(@PathVariable String gameId, Principal principal){
+        try{
+            return ResponseEntity.ok(gameService.getOneGameToEdit(gameId, principal));
+        } catch (NoSuchElementException e){
+            return new ResponseEntity(e.getMessage(), HttpStatus.NOT_FOUND);
+        }
+    }
+
+    @PutMapping("/edit")
+    public ResponseEntity<Void> editOneGame(@RequestBody Game game, Principal principal){
+        try {
+            gameService.editOneGame(game, principal);
```

Comment:
Same as above for principal.
editOneGame can also throw NoSuchElementExceptions which are not handled by controller